File: lib/admin/features/monitoring/services/admin_monitoring_service.dart
import '../../../../core/services/firebase_service.dart';

class CollectionHealth {
  final String name;
  final int count;
  const CollectionHealth(this.name, this.count);
}

/// Raw operational counts, distinct from the Dashboard's business-framed
/// metrics — this is "is data flowing into every collection", not "how is
/// the business doing".
class AdminMonitoringService {
  AdminMonitoringService._();
  static final AdminMonitoringService instance = AdminMonitoringService._();

  static const _collections = [
    'users',
    'tickets',
    'openTickets',
    'jobs',
    'payments',
    'ratings',
    'reports',
    'safetyAlerts',
    'deviceTokens',
    'announcements',
    'admins',
    'auditEvents',
  ];

  Future<List<CollectionHealth>> load() async {
    final db = FirebaseService.instance.firestore;
    final results = await Future.wait(
      _collections.map((name) => db.collection(name).count().get()),
    );
    return [
      for (var i = 0; i < _collections.length; i++)
        CollectionHealth(_collections[i], results[i].count ?? 0),
    ];
  }
}

File: admin-web/src/services/adminMonitoringService.ts
import { collection, getCountFromServer } from 'firebase/firestore'
import { db } from '@/lib/firebase'
import type { CollectionHealth } from '@/types/models'

/** Raw operational counts, distinct from the Dashboard's business-framed metrics — "is data flowing", not "how is the business doing". */
const monitoredCollections = [
  'users', 'tickets', 'openTickets', 'jobs', 'payments', 'ratings',
  'reports', 'safetyAlerts', 'deviceTokens', 'announcements', 'admins', 'auditEvents',
]

export async function loadCollectionHealth(): Promise<CollectionHealth[]> {
  const results = await Promise.all(
    monitoredCollections.map((name) => getCountFromServer(collection(db, name))),
  )
  return monitoredCollections.map((name, i) => ({ name, count: results[i]!.data().count }))
}
